proposals/records-wip.md

```diff
@@ -89,13 +91,16 @@ A `with` expression allows for "non-destructive mutation", designed to
 produce a copy of the receiver expression with modifications to properties
 listed in the `anonymous_object_initializer`.
 
-A valid `with` expression has a receiver with a non-void type. The receiver type must contain an accessible parameterless instance method called `Clone` whose return type must be the type of the receiver express type, or a base type thereof.
+A valid `with` expression has a receiver with a non-void type. The receiver type must contain an accessible
+parameterless instance method called `Clone` whose return type must be the receiver type, or a base type thereof.
 
 On the right hand side of the `with` expression is an `anonymous_object_initializer` with a
-sequence of assignments with a compiler-generated record property of the receiver on the left-hand side of the
-assignment, and an arbitrary expression on the right-hand side which is implicitly convertible to the type
-of the left-hand side.
+sequence of assignments, each with an `initonly` property (see [Properties](#Properties)) of the `Clone`
+return type on the left-hand side of the assignment, and an arbitrary expression on the right-hand side which is 
+implicitly convertible to the type of the property.
+
+The evaluation of a `with` expression calls the `Clone` method exactly once,
+and then sets the backing field of each `initonly` property in the argument list to 
```

Comment:
```suggestion
and then sets the backing field of each `init` property in the argument list to 
```
And perhaps it should be calling the `init` accessor instead of the backing field?